UIforETW/Utility.cpp

```diff
@@ -19,7 +19,166 @@ limitations under the License.
 #include <fstream>
 #include <direct.h>
 
-std::vector<std::wstring> split(const std::wstring& s, char c)
+
+namespace {
+PCWSTR const WPAStartupFileName = L"\\Startup.wpaProfile";
+
+std::wstring GetDocumentsFolderPath()
+{
+	//must CoTaskMemFree when done!
+	PWSTR docsPathTemp = NULL;
+	const HRESULT docsPathResult = ::SHGetKnownFolderPath(FOLDERID_Documents, KF_FLAG_NO_ALIAS, NULL, &docsPathTemp);
+	if (FAILED(docsPathResult))
+	{
+		if (docsPathResult == E_INVALIDARG)
+		{
+			debugPrintf(L"SHGetKnownFolderPath (for Documents) failed to retrieve the path with E_INVALIDARG\n");
+			std::terminate();
+		}
+		else if (docsPathResult == E_FAIL)
+		{
+			debugPrintf(L"SHGetKnownFolderPath (for Documents) failed to retrieve the path with E_FAIL\n");
+		}
+		else
+		{
+			debugPrintf(L"SHGetKnownFolderPath (for Documents) failed to retrieve the path with an unknown error: 0x%08I32x \n", docsPathResult);
+		}
+		UIETWASSERT(docsPathTemp == NULL);
+		return L"";
+	}
+	std::wstring docsPath(docsPathTemp);
+	CoTaskMemFree(docsPathTemp);
+	return docsPath;
+}
+
+void copyWPAProfileToDocuments(const bool force)
+{
+	// First copy the WPA 8.1 startup.wpaProfile file
+	std::wstring docsPath(GetDocumentsFolderPath());
+
+	if (force)
+		outputPrintf(L"\n");
+
+	if (docsPath.empty())

```

Comment:
It's actually no longer relevant now that I've removed the error path from `GetDocumentsFolderPath`.